src/actions_eval.ts

```diff
@@ -26,12 +26,17 @@ export async function eval_for_client(code: string, args: any): Promise<any> {
 
     void args;
 
-    const value_type = typeof result;
-    if (value_type === "object") {
-        return String(result);
-    } else if (value_type === "function") {
-        return `[Function: ${result.name}]`;
-    } else {
-        return result;
+    if (typeof result === "object") {
```

Comment:
Might it make sense to remove the conditional and change this to something like

```ts
let data: string | undefined;
try {
     data = JSON.stringify(result);
} catch (e) { /* ... */ }

if (data === undefined)  {
  throw new Error(`Invalid data from eval data: ${result}`)
}

return JSON.decode(data); // or maybe even just `return result`;
```

This ensures that the data we return to Lua is serializable as JSON, and avoids the case your comment warns about

Reply:
> or maybe even just `return result`, since we know the data can safely be serialized.

For
```js
const obj = { a() {} }
```
- JSON.stringify: ✔
- neovim transform: ❌

Directly returning the result will fail for commonly encountered objects like TextEditor and Document.

Reply:
edge cases like that would be helpful to note in the commit message